src/mmw/apps/modeling/mapshed.py

```diff
@@ -57,6 +57,9 @@ def nearest_weather_stations(geom, n=NUM_WEATHER_STATIONS):
     with connection.cursor() as cursor:
         cursor.execute(sql, [geom.wkt, n])
 
+        if cursor.rowcount == 0:
+            raise Exception("No weather stations found.")
+
```

Comment:
I can see how the closest weather station might be far away, but how can there be zero weather stations near a geom?